src/wasm-interpreter.h

```diff
@@ -2228,6 +2228,186 @@ class ModuleInstance
     : ModuleInstanceBase(wasm, externalInterface) {}
 };
 
+static const Name NONSTANDALONE_FLOW("Binaryen|nonstandalone");
+
+typedef std::unordered_map<LocalGet*, Literals> GetValues;
+
+// Evaluates a standalone expression. Errors if we hit anything that can't be
+// evaluated.
+class StandaloneExpressionRunner
+  : public ExpressionRunner<StandaloneExpressionRunner> {
+  Module* module;
+
+  // map gets to constant values, if they are known to be constant
+  GetValues& getValues;
+  // map local indexes to values set in the context of this runner
+  std::unordered_map<Index, Literals> setLocalValues;
+  // map global names to values set in the context of this runner
+  std::unordered_map<Name, Literals> setGlobalValues;
+
+public:
+  // Whether we are trying to precompute down to an expression (which we can
+  // do on say 5 + 6) or to a value (which we can't do on a local.tee that
+  // flows a 7 through it). When we want to replace the expression, we can
+  // only do so when it has no side effects. When we don't care about
+  // replacing the expression, we just want to know if it will contain a known
+  // constant.
+  enum Intent {
+    // Intent is to evaluate the expression, so we can ignore some side effects
```

Comment:
`some` seems to imply that certain side effects are ok but others are not, or that there is some amount of them that is ok. maybe remove that word?

Reply:
The `some` was intended, in that for example a `local.tee` is ok, but branching or a trap of course is not. Perhaps I should make that more clear?

Reply:
Made this more clear in the comments